metadata_please/source_checkout.py

```diff
@@ -262,6 +403,31 @@ def from_setup_py_checkout(path: Path) -> bytes:
             for i in deps:
                 buf.append("Requires-Dist: " + merge_extra_marker(extra_name, i) + "\n")
 
+    n = v.setup_call_args.get("name")
+    if n:
+        if n is UNKNOWN:
+            raise ValueError("Complex setup call can't extract name")
```

Comment:
I think long-term these shouldn't raise; the snippets above do that because finding deps was the original use of this.  It doesn't matter for your artifact-validation use case though.